server/src/main/java/org/elasticsearch/index/search/QueryParserHelper.java

```diff
@@ -149,7 +149,12 @@ private QueryParserHelper() {}
                     // other exceptions are parsing errors or not indexed fields: keep
                 }
             }
-            fields.put(fieldName, weight);
+            // handle duplicates
+            float w = weight;
+            if (fields.containsKey(fieldType.name())) {
+                w *= fields.get(fieldType.name());
```

Comment:
Should we document this behaviour?
Also can we have a test that shows this change in the weight?

Reply:
I pushed a dedicated test and fixed the merging of boosts in https://github.com/elastic/elasticsearch/pull/42328/commits/72ff29af57c1e9981d7e34651c43d87bfcffe4d7. 

> Should we document this behaviour?

The behavior should be the same as before but instead of having two clauses with different boosts (one for the alias field and one for the concrete field) we'll have a single clause with the merged boost. Scoring wise this should be identical.